diff --git a/vendor/rails/activesupport/lib/active_support/cache.rb b/vendor/rails/activesupport/lib/active_support/cache.rb
index 070ba12..f17687e 100644
--- a/vendor/rails/activesupport/lib/active_support/cache.rb
+++ b/vendor/rails/activesupport/lib/active_support/cache.rb
@@ -150,7 +150,8 @@ module ActiveSupport
       #   cache.fetch("foo")  # => nil
       def fetch(key, options = {})
         @logger_off = true
-        if !options[:force] && value = read(key, options)
+        if !options[:force] && exist?(key, options)
+          value = read(key, options)
           @logger_off = false
           log("hit", key, options)
           value
